src/csharp/Microsoft.Spark/Sql/SparkSession.cs

```diff
@@ -389,6 +390,42 @@ public void Dispose()
         /// </summary>
         public void Stop() => _jvmObject.Invoke("stop");
 
+        /// <summary>
+        /// Get the <see cref="VersionSensor.VersionInfo"/> for the Microsoft.Spark assembly
+        /// running on the Spark Driver and make a "best effort" attempt in determining the version
+        /// of Microsoft.Spark.Worker assembly on the Spark Executors.
+        /// </summary>
+        /// <returns>
+        /// A <see cref="DataFrame"/> containing the <see cref="VersionSensor.VersionInfo"/>
+        /// </returns>
+        public DataFrame Version(int numPartitions = 10)
```

Comment:
Just curious, why have we chosen `numPartitions` as 10 here?

Reply:
Any number is fine, just some safe default to run for local / small clusters.  Do you have a recommendation on what value we can use ?

Reply:
No 10 is fine, I was just wondering if there was a specific reason.

Reply:
Do we want to add a test covering this since it is a public API?

Reply:
Also I am wondering if asking for `numPartitions` is really necessary for finding the version from the user perspective. Would having it as a constant inside the function pose a problem?

Reply:
What would be the pros and cons of hardcoding the numbers instead ?

Reply:
Pros I think would be usability, since it might be hard to intuitively understand what number of partitions a user should enter and what it has to do with getting version information. Cons, I guess include not being able to trigger every worker node and if some nodes have a different worker version on them then that would go uncaught. But that could happen even with taking `numPartitions` as an argument so not sure what the benefit there is, unless we document it in more depth so as to advise the user of the importance of estimating a 'correct' `numPartitions` value, maybe explain in detail what `best effort` means etc. Thoughts @imback82 ?